samples/tcpsrv.lua

```diff
@@ -0,0 +1,54 @@
+#! /usr/bin/env lua
+
+--
+-- Sample TCP echo server. Listens on 3333 port
+-- Use netcat to send strings to server (ncat 127.0.0.1 3333)
+--
+
+local lgi = require 'lgi'
+local Gio = lgi.Gio
+
+local app = Gio.Application{application_id = 'org.test.tcptest',
+                            flags = Gio.ApplicationFlags.NON_UNIQUE}
+
+local service = Gio.SocketService.new()
+service:add_address(
+   Gio.InetSocketAddress.new_from_string("127.0.0.1", 3333),
+   Gio.SocketType.STREAM,
+   Gio.SocketProtocol.TCP)
+
+local function get_message(conn, istream, ostream)
+   ostream:async_write("> ")
+   local bytes = istream:async_read_bytes(4096)
```

Comment:
All the functions calls here ignore error. `:async_read_bytes()` should actually return two values: the `bytes` and an instance of `GError`. I am not quite sure about the details (is `bytes == nil` when an error is returned? I think so...).
Also: No idea how errors should be handled here.

The same "ignores errors" comment applies to almost all the functions in both examples.

Reply:
I *think* (but am not sure) that `assert(foo)` could be used for error handling everywhere.